### PR TITLE
fix: validation example

### PR DESCRIPTION
**Pull request by lord007tn**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
I changed the `validation-example.js` to pass the JSON validation schema.
and add a section in the `readme.md` file so people can have a quick access to make it work
Fixes https://github.com/fastify/fastify-multipart/issues/173

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -290,6 +290,46 @@ If you try to read from a stream and pipe to a new file, you will obtain an empt
 
 ## JSON Schema body validation
 
+You can use `fastify-multipart` with schema validation, this is a full working example.
```
Comment (Eomm) [inline README.md:293]: Could you add a test for this new example?

We can make it consistent in this way

**Comment by lord007tn** [comment]
Still working on It 
further in the project the Ajv validators seems to make fastify fail in some edge cases 
I'm trying to find solutions and I will update it soon 
with tests also

**Comment by lord007tn** [comment]
I added a valid example that worked for me 
but didn't know how to implement unit testing :/

I will give it a shot when I have more time
you could verify it

**Comment by lord007tn** [comment]
I added a working example that i've tried in my work project 
and it works fine 
still lacks unit testing 
i would give it a try when i have more time 
you could verify it

File: README.md [hunk 2]
```diff
@@ -372,6 +374,67 @@ The shared schema, that is added, will look like this:
 }
 ```
 
+### JSON Schema with Swagger
+
+If you want to use `@fastify/multipart` with `@fastify/swagger` and `@fastify/swagger-ui` you must add a new type called `isFiled` and use custom instance of validator compiler [Docs](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#validator-compiler).
```
Comment (fertolg) [inline README.md:379]:
```suggestion
If you want to use `@fastify/multipart` with `@fastify/swagger` and `@fastify/swagger-ui` you must add a new type called `isFile` and use custom instance of validator compiler [Docs](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#validator-compiler).
```

File: README.md [hunk 3]
```diff
@@ -297,6 +297,7 @@ fastify.register(require('@fastify/multipart'), { attachFieldsToBody: 'keyValues
 
 fastify.post('/upload/files', {
   schema: {
+    consumes: ["multipart/form-data"],
```
Comment (aj-gameon) [inline README.md:300]:
```suggestion
    consumes: ['multipart/form-data'],
```

File: README.md [hunk 4]
```diff
@@ -331,6 +332,7 @@ fastify.register(require('@fastify/multipart'), opts)
...
```
Comment (aj-gameon) [inline README.md:335]:
```suggestion
    consumes: ['multipart/form-data'],
```

File: README.md [hunk 5]
```diff
@@ -463,6 +527,7 @@ fastify.post('/upload/files', {
 })
 ```
 
+
```
Comment (aj-gameon) [inline README.md:530]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -433,6 +496,7 @@ fastify.register(require('@fastify/multipart'), opts)
 
 fastify.post('/upload/files', {
   schema: {
+    consumes: ["multipart/form-data"],
```
Comment (aj-gameon) [inline README.md:499]:
```suggestion
    consumes: ['multipart/form-data'],
```

File: examples/example-with-swagger.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  keyword: 'isFile',
+  compile: (_schema, parent, _it) => {
+    parent.type = 'file'
+    delete parent.isFile
+    return () => true
+  }
+})
+fastify.setValidatorCompiler(({ schema, method, url, httpPart }) => {
+  return ajv.compile(schema)
+})
```
Comment (fertolg) [inline examples/example-with-swagger.js:31]: If you go this route, then any additional schemas will need to be added to the validator instead of the normal `fastify.addSchema` usage, which may not be intuitive. See [the note in the validator compiler docs](https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#validator-compiler:~:text=Note%3A%20If%20you%20use%20a%20custom%20instance%20of%20any%20validator%20(even%20Ajv)%2C%20you%20have%20to%20add%20schemas%20to%20the%20validator%20instead%20of%20Fastify%2C%20since%20Fastify%27s%20default%20validator%20is%20no%20longer%20used%2C%20and%20Fastify%27s%20addSchema%20method%20has%20no%20idea%20what%20validator%20you%20are%20using.).

Instead of defining an entirely custom validator, would passing the custom keyword as options when initializing fastify allow users to still rely on Fastify's private AJV implementation?

- See [Fastify's AJV factory docs](https://www.fastify.io/docs/latest/Reference/Server/#factory-ajv)
- See the [AJV keywords option](https://ajv.js.org/options.html#keywords)

**Comment by fertolg** [comment]
FWIW, I've been trying to use this instructions in my project for half of today, and I don't think this actually validates properly. If I follow the example I am able to see a file button in Swagger UI, but when attempting to make a request it fails validation with `Error: body must be object`. It's possible I'm missing something else or my specific use case is conflicting with these instructions, are you able to actually make a request using the swagger UI, and the request's body validates properly?

File: examples/example-with-swagger.js (hunk 7 above)
Comment (lord007tn) [reply examples/example-with-swagger.js:31]: you made a good point 
as myself I didn't use `addSchema()` but it's valid I will push changes

**Review by mcollina — APPROVED** [review]
lgtm
